app/components/pill/Pill.tsx

```diff
@@ -111,8 +111,9 @@ const Pill = () => {
           setIsManualRecording(false)
           isManualRecordingRef.current = false
           setRecordingMode(undefined)
+          // Only clear volume history when recording stops
+          setVolumeHistory([])
```

Comment:
Now users can switch modes for ito mid-stream, but the audio bars will not be cleared. They'll simply change color